src/plugins/ui-plugin.ts

```diff
@@ -18,7 +18,10 @@ export default class UIPlugin extends EventEmitter {
 
     render() {}
 
-    setSettings(...args) {}
+    /* eslint-disable-next-line @typescript-eslint/no-unused-vars --
+     unused argument, to keep signature compatible with children
+     */
+    setSettings(settings) {}
```

Comment:
You can rewrite this by function signature, like this
```js
abstract setSettings(...settings: any[]): void
```
And make this class abstract.

Reply:
> You can rewrite this by function signature, like this
> 
> ```js
> abstract setSettings(...settings: any[]): void
> ```
> 
> And make this class abstract.

tnx, I am not really an OOP guy, for long time working only with functional/procedural programming paradigm
this project is a good refresher on the OOP :D

Reply:
Hmm.
Looks like you can make this method optional in the abstract class.
```ts
abstract class Base {
    abstract required(): void;
    optional?(): string;

    print(): void {
        console.log(this.optional?.() ?? "Base");
    }
}

class Child1 extends Base {
    required(): void { }
}

class Child2 extends Base {
    required(): void { }
    optional(): string {
        return "Child";
    }
}

```